specification/components/schemas/extensions/Extension-eRS-ReferralState.yaml

```diff
@@ -0,0 +1,99 @@
+    example: 'https://fhir.nhs.uk/STU3/StructureDefinition/Extension-eRS-ReferralState-1'
+  extension:
+    type: array
+    description: |
+      | Field                                      | Cardinality | Notes                                                                            |
+      | ------------------------------------------ | ----------- | -------------------------------------------------------------------------------- |
+      | `state`                                    | `1..1`      | Outlines the current state of the `ReferralRequest`                              |
+      | `reason`                                   | `0..1`      | Details the reason why the `ReferralRequest` is in this state, where appropriate |
+      | `comments`                                 | `0..1`      | Any comments around the current `ReferralRequest` state                          |
+    items:
+      anyOf:
+        - type: object
+          required:
+            - url
+            - valueCoding
+          properties:
+            url:
+              type: string
+              enum:
+                - state
+              example: 'state'
+            valueCoding:
+              type: object
+              required:
+                - system
+                - code
+                - display
+              properties:
+                system:
+                  type: string
+                  enum:
+                    - https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReferralState-1
+                  example: https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReferralState-1
+                code:
+                  type: string
+                  enum:
+                    # This is just a selection of potential defining states. The full list is still TBC
+                    - AWAITING_TRIAGE
+                    - BOOKED
+                    - UNBOOKED
+                  example: 'AWAITING_TRIAGE'
+                display:
+                  type: string
+                  enum:
+                    - 'Awaiting Triage'
+                    - 'Booked'
+                    - 'Unbooked'
+                  example: 'Awaiting Triage'
+        - type: object
+          required:
+            - url
+            - valueCoding
+          properties:
+            url:
+              type: string
+              enum:
+                - reason
+              example: 'reason'
+            valueCoding:
+              type: object
+              required:
+                - system
+                - code
+              properties:
+                system:
+                  type: string
+                  enum:
+                    - __baseurl__/FHIR/STU3/CodeSystem/APPOINTMENT-CANCELLATION-REASON
+                    - __baseurl__/FHIR/STU3/CodeSystem/REFERRAL-CANCELLATION-REASON
+                    - https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReviewOutcome-1
+                    - https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReviewReason-1
```

Comment:
I suspect there are more possible systems allowed here too, but they can be added later when finalised.